src/ring_theory/polynomial/content.lean

```diff
@@ -195,25 +196,68 @@ begin
     apply h _ (C_content_dvd _) }
 end
 
-lemma eq_C_mul_primitive (p : polynomial R) :
-  ∃ (r : R) (q : polynomial R), p = C r * q ∧ q.is_primitive ∧ p.nat_degree = q.nat_degree :=
+section prim_part
+variable [decidable_eq R]
```

Comment:
Why the `decidable_eq` assumption? Your definition is `noncomputable` anyway, so I would rather `open_locale classical`.